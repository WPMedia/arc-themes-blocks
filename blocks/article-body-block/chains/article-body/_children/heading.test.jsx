const React = require("react");
const { mount } = require("enzyme");

describe("the article body Heading component", () => {
	it("should render the correct heading", () => {
		const headingData = {
			_id: "CF5ARXXK6BHJ5LO45DZCCBHL7U",
			type: "header",
			level: 3,
			additional_properties: {
				comments: [],
				inline_comments: [],
				_id: 1563473120776,
			},
			content:
				'Heading 3 - <b>bold</b> <i>italic</i> <u>underline</u> <a href="https://www.washingtonpost.com/" target=_blank>hyperlink</a>',
		};
		const { default: Heading } = require("./heading");
		const wrapper = mount(<Heading element={headingData} />);
		expect(wrapper.find("h3").length).toBe(1);
		expect(wrapper.find("h3").text()).toMatch("Heading 3 - bold italic underline hyperlink");
		expect(wrapper.find("h3").html()).toMatchInlineSnapshot(
			'"<h3 class=\\"sc-bdVaJa drnjHs\\">Heading 3 - <b>bold</b> <i>italic</i> <u>underline</u> <a href=\\"https://www.washingtonpost.com/\\" target=\\"_blank\\">hyperlink</a></h3>"'
		);
	});

	it("should default to h2 if no heading level is given", () => {
		const headingData = {
			_id: "CF5ARXXK6BHJ5LO45DZCCBHL7U",
			type: "header",
			additional_properties: {
				comments: [],
				inline_comments: [],
				_id: 1563473120776,
			},
			content:
				'Heading 3 - <b>bold</b> <i>italic</i> <u>underline</u> <a href="https://www.washingtonpost.com/" target=_blank>hyperlink</a>',
		};
		const { default: Heading } = require("./heading");
		const wrapper = mount(<Heading element={headingData} />);
		expect(wrapper.find("h2").length).toBe(1);
		expect(wrapper.find("h2").html()).toMatchInlineSnapshot(
			'"<h2 class=\\"sc-bdVaJa drnjHs\\">Heading 3 - <b>bold</b> <i>italic</i> <u>underline</u> <a href=\\"https://www.washingtonpost.com/\\" target=\\"_blank\\">hyperlink</a></h2>"'
		);
		expect(wrapper.find("h2").text()).toMatch("Heading 3 - bold italic underline hyperlink");
	});
});
