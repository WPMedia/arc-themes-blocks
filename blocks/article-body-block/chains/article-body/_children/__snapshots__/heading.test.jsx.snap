// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the article body Heading component should default to h2 if no heading level is given 1`] = `"<h2 class=\\"sc-bdVaJa eajvwU\\">Heading 3 - <b>bold</b> <i>italic</i> <u>underline</u> <a href=\\"https://www.washingtonpost.com/\\" target=\\"_blank\\">hyperlink</a></h2>"`;

exports[`the article body Heading component should render the correct heading 1`] = `"<h3 class=\\"sc-bdVaJa eajvwU\\">Heading 3 - <b>bold</b> <i>italic</i> <u>underline</u> <a href=\\"https://www.washingtonpost.com/\\" target=\\"_blank\\">hyperlink</a></h3>"`;
